presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/raw/RawRowEncoder.java

```diff
@@ -0,0 +1,282 @@
+    }
+
+    private FieldType parseFieldType(EncoderColumnHandle columnHandle)
+    {
+        try {
+            return Optional.ofNullable(columnHandle.getDataFormat())
+                    .map(dataFormat -> FieldType.valueOf(dataFormat.toUpperCase(Locale.ENGLISH)))
+                    .orElse(FieldType.BYTE);
+        }
+        catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException(format("Invalid dataFormat '%s' for column '%s'", columnHandle.getDataFormat(), columnHandle.getName()));
+        }
+    }
+
+    private void checkFieldType(EncoderColumnHandle columnHandle, FieldType fieldType)
+    {
+        String columnName = columnHandle.getName();
+        Type columnType = columnHandle.getType();
+        if (columnType == BIGINT) {
+            checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.BYTE, FieldType.SHORT, FieldType.INT, FieldType.LONG);
+        }
+        else if (columnType == INTEGER) {
+            checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.BYTE, FieldType.SHORT, FieldType.INT);
+        }
+        else if (columnType == SMALLINT) {
+            checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.BYTE, FieldType.SHORT);
+        }
+        else if (columnType == TINYINT) {
+            checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.BYTE);
+        }
+        else if (columnType == BOOLEAN) {
+            checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.BYTE, FieldType.SHORT, FieldType.INT, FieldType.LONG);
+        }
+        else if (columnType == DOUBLE) {
+            checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.DOUBLE, FieldType.FLOAT);
+        }
+        else if (isVarcharType(columnType)) {
+            checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.BYTE);
+        }
+    }
+
+    private void checkFieldTypeOneOf(FieldType declaredFieldType, String columnName, Type columnType, FieldType... allowedFieldTypes)
+    {
+        if (!Arrays.asList(allowedFieldTypes).contains(declaredFieldType)) {
+            throw new IllegalArgumentException(format(
+                    "Wrong dataFormat '%s' specified for column '%s'; %s type implies use of %s",
+                    declaredFieldType.name(),
+                    columnName,
+                    columnType.getDisplayName(),
+                    Joiner.on("/").join(allowedFieldTypes)));
+        }
+    }
+
+    private int parseMapping(EncoderColumnHandle columnHandle, FieldType fieldType)
+    {
+        String mapping = Optional.ofNullable(columnHandle.getMapping()).orElse("0");
+        Matcher mappingMatcher = MAPPING_PATTERN.matcher(mapping);
+        if (!mappingMatcher.matches()) {
+            throw new IllegalArgumentException(format("Invalid mapping format '%s' for column '%s'", mapping, columnHandle.getName()));
+        }
+        int start = parseInt(mappingMatcher.group(1));
```

Comment:
nit: possibly catch `NumberFormatException` and wrap in exception with clear message. It can throw if value provided does not fit in int.